### DISCUSSION TITLE
Add Support for the Model Context Protocol

### DISCUSSION THREAD

**Discussion by asaikali** [open-webui/open-webui#7431] [Ideas]
The Model Context Protocol https://modelcontextprotocol.io/introduction provides a generic way for a GUI such as open-webui to pull in arbitrary data into the chat. MCP was open sourced by Anthropic, the user experience of MCP + Claude Desktop app is very nice try the quick start at https://modelcontextprotocol.io/quickstart

It would be awesome if MCP was fully supported by OpenWeb UI then folks can bring their own MCP servers and connect them to OpenWeb UI and have a very rich experience.

**Comment by jkuehn** [comment]
I agree. See also #7363.

**Comment by FellowTraveler** [comment]
Definitely interested in this.

**Comment by kevinkre** [comment]
What's the delay? People are hemorraging off the platform to librechat. Do you care?

**Reply by SiNZeRo** [reply]
up

**Comment by longregen** [comment]
This is becoming super interesting. The standard that tool use was needing.

**Comment by davideuler** [comment]
Any progress on MCP protocol support? Also it maybe a duplicate design of function/pipelines.
The whole ecosystem has been growing fast for MCP. If it is integrated into open-webui, it would be super easy for us to leverage lots of related work from MCP servers.

**Comment by julianwyngaard** [comment]
Supporting MCP will be a super valuable capability/feature and seems to be the way forward as a standard. 

People can add custom capabilities to Open WebUI chats by creating and use their own tools using any language they want, or use any MCP server they like, and not necessarily have to be doing things in python with functions/pipelines, not everyone is a python developer ✋. Really hoping this makes its way into Open WebUI soon!

**Comment by tjbck** [comment]
https://github.com/open-webui/open-webui/discussions/7363
